vendor/bat-native-ads/src/bat/ads/internal/covariates/covariate_manager.cc

```diff
@@ -29,83 +29,90 @@ using UserActivityEventToCovariateTypesMapping =
-        {brave_federated::mojom::CovariateType::
-             kNumberOfClickedBackOrForwardNavigationButtonsEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastClickedBackOrForwardNavigationButtonsEvent}},
-       {UserActivityEventType::kClickedLink,
-        {brave_federated::mojom::CovariateType::kNumberOfClickedLinkEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastClickedLinkEvent}},
-       {UserActivityEventType::kClickedReloadButton,
-        {brave_federated::mojom::CovariateType::
-             kNumberOfClickedReloadButtonEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastClickedReloadButtonEvent}},
-       {UserActivityEventType::kClosedTab,
-        {brave_federated::mojom::CovariateType::kNumberOfClosedTabEvents,
-         brave_federated::mojom::CovariateType::kTimeSinceLastClosedTabEvent}},
-       {UserActivityEventType::kTabChangedFocus,
-        {brave_federated::mojom::CovariateType::
-             kNumberOfFocusedOnExistingTabEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastFocusedOnExistingTabEvent}},
-       {UserActivityEventType::kNewNavigation,
-        {brave_federated::mojom::CovariateType::kNumberOfNewNavigationEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastNewNavigationEvent}},
-       {UserActivityEventType::kOpenedNewTab,
-        {brave_federated::mojom::CovariateType::kNumberOfOpenedNewTabEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastOpenedNewTabEvent}},
-       {UserActivityEventType::kTabStartedPlayingMedia,
-        {brave_federated::mojom::CovariateType::kNumberOfPlayedMediaEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastPlayedMediaEvent}},
-       {UserActivityEventType::kSubmittedForm,
-        {brave_federated::mojom::CovariateType::kNumberOfSubmittedFormEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastSubmittedFormEvent}},
-       {UserActivityEventType::kTypedAndSelectedNonUrl,
-        {brave_federated::mojom::CovariateType::
-             kNumberOfTypedAndSelectedNonUrlEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastTypedAndSelectedNonUrlEvent}},
-       {UserActivityEventType::kTypedKeywordOtherThanDefaultSearchProvider,
-        {brave_federated::mojom::CovariateType::
-             kNumberOfTypedKeywordOtherThanDefaultSearchProviderEvents,
-         brave_federated::mojom::CovariateType::
-             kTimeSinceLastTypedKeywordOtherThanDefaultSearchProviderEvent}},
-       {UserActivityEventType::kTypedUrl,
-        {brave_federated::mojom::CovariateType::kNumberOfTypedUrlEvents,
-         brave_federated::mojom::CovariateType::kTimeSinceLastTypedUrlEvent}}});
-  return *mappings;
+  static const base::NoDestructor<UserActivityEventToCovariateTypesMapping>
```

Comment:
Need to consider converting to fixed_flat_map

Reply:
please do

Reply:
@aseren will fix in his related PR